src/Symfony/Component/Validator/Constraints/TypeValidator.php

```diff
@@ -37,14 +71,14 @@ public function validate($value, Constraint $constraint)
 
         foreach ($types as $type) {
             $type = strtolower($type);
-            $type = 'boolean' === $type ? 'bool' : $type;
-            $isFunction = 'is_'.$type;
-            $ctypeFunction = 'ctype_'.$type;
-            if (\function_exists($isFunction) && $isFunction($value)) {
-                return;
-            } elseif (\function_exists($ctypeFunction) && $ctypeFunction($value)) {
-                return;
-            } elseif ($value instanceof $type) {
+            if (\array_key_exists($type, self::VALIDATION_FUNCTIONS)) {
```

Comment:
```php
if (null !== ($function = self::VALIDATION_FUNCTIONS[$type] ?? null) && $function($value)) {
    return;
}
```

Reply:
Hi @nicolas-grekas. I only changed `\array_key_exists` with `isset` so far.

Did a small test locally that seems to be visible here as well: https://3v4l.org/pT0lC
and, interestingly, the current version:
```php
if (isset(self::VALIDATION_FUNCTIONS[$type])) {
    $validationFunction = self::VALIDATION_FUNCTIONS[$type];
    if ($validationFunction($value)) {
        return;
    }
}

```
is faster than
```php
if (isset(self::VALIDATION_FUNCTIONS[$type]) && (self::VALIDATION_FUNCTIONS[$type])($value)) {
    return;
}
```
which is also a bit faster than the one you proposed
```php
if (null !== ($function = self::VALIDATION_FUNCTIONS[$type] ?? null) && $function($value)) {
    return;
}
```

Personally, I like a more detailed and readable version of the code so the maintenance cost will stay low over time but I assume small speed improvements are something to look at here.
Of course, what "more readable" means is purely subjective.
Have a look again please and let me know if I should keep this current version or change it to one of the other 2 options that are one-liners.
I'm good with any version as I don't know very what practices are used for symfony in terms of speed and readability. I'm happy enough that I learned something.
Thanks!